tests/contest/runtimetest/src/tests.rs

```diff
@@ -550,6 +552,51 @@ pub fn test_io_priority_class(spec: &Spec, io_priority_class: IOPriorityClass) {
     }
 }
 
+pub fn test_validate_root_readonly(spec: &Spec) {
+    let root = spec.root().as_ref().unwrap();
+    if root.readonly().unwrap() {
+        if let Err(e) = test_dir_write_access("/") {
+            let errno = Errno::from_raw(e.raw_os_error().unwrap());
+            if errno == Errno::EROFS {
+                /* This is expected */
+            } else {
+                eprintln!(
+                    "readonly root filesystem, error in testing write access for path /, error: {}",
+                    errno
+                );
+            }
+        }
+        if let Err(e) = test_dir_read_access("/") {
+            if let Some(errno_code) = e.raw_os_error() {
+                let errno = Errno::from_raw(errno_code);
+                eprintln!(
+                    "readonly root filesystem, but error in testing read access for path /, error: {}",
+                    errno
+                );
+            }
+        }
+    } else {
```

Comment:
```suggestion
        if let Err(e) = test_dir_read_access("/") {
            eprintln!(
                "readonly root filesystem, but error in testing read access for path /, error: {}",
                e
            );
    } else {
```
As I had mentioned in my previous comment,  we don't need specific check for raw_os_error. As long as this is error, the test is failing. Similar for the  two cases below this. Also I think here, you are converting the error to raw os error, and then using errorno to convert it back into the original error.

Reply:
Sorry for any inconvenience caused by my misunderstanding.
I fixed this. Is this ok?

https://github.com/youki-dev/youki/pull/2976/commits/736127756158af9dd9328f8bb77cb08975f78858

Reply:
Yep, this is what I wanted :+1: